mono/metadata/w32process-unix.c

```diff
@@ -1913,25 +1911,28 @@ process_create (const gunichar2 *appname, const gunichar2 *cmdline,
 	 * new process inherits the same environment.
 	 */
 	if (process_info->env_variables) {
-		gint i;
-		MonoString *var;
+		MonoArrayHandle array = MONO_HANDLE_NEW (MonoArray, process_info->env_variables);
```

Comment:
I see `MonoW32ProcessInfo` is a managed struct that's been pinned and passed by ref.

Reply:
Not quite. I pinned its constituents.

Reply:
Yes - this seems fine - the managed ref constituents need to be accessed either by pinning or via handles, as you've done.  I was just surprised to see this use of `MONO_HANDLE_NEW` - I don't think we often pass struct like this.

Reply:
Imho we should revise the managed/native interface here, but it isn't performance sensitive and there are many fish to fry. Remove the two structs and pass the constituents imho, `fixed` the strings, `handles()` the array, ref the two actual out  parameters, pass the other values directly, etc.